dask_match/core.py

```diff
@@ -1,628 +1,279 @@
-        else:
-            return len(self.divisions) - 1
-
-    @property
-    def _name(self):
-        if "_name" in self._parameters:
-            idx = self._parameters.index("_name")
-            return self.operands[idx]
-        return funcname(type(self)).lower() + "-" + tokenize(*self.operands)
-
-    @property
-    def columns(self):
-        if "columns" in self._parameters:
-            idx = self._parameters.index("columns")
-            return self.operands[idx]
-        else:
-            return self._meta.columns
-
-    @property
-    def dtypes(self):
-        return self._meta.dtypes
+    def expr(self):
+        return self.__expr
 
     @property
     def _meta(self):
-        if "_meta" in self._parameters:
-            idx = self._parameters.index("_meta")
-            return self.operands[idx]
-        raise NotImplementedError()
-
-    def _divisions(self):
-        raise NotImplementedError()
+        return self.expr._meta
 
     def __dask_graph__(self):
-        """Traverse expression tree, collect layers"""
-        stack = [self]
-        seen = set()
-        layers = []
-        while stack:
-            expr = stack.pop()
-
-            if expr._name in seen:
-                continue
-            seen.add(expr._name)
-
-            layers.append(expr._layer())
-            for operand in expr.operands:
-                if isinstance(operand, Expr):
-                    stack.append(operand)
-
-        return toolz.merge(layers)
+        return self.expr.__dask_graph__()
 
     def __dask_keys__(self):
-        return [(self._name, i) for i in range(self.npartitions)]
+        return self.expr.__dask_keys__()
+
+    @property
+    def dask(self):
+        return self.__dask_graph__()
 
     def __dask_postcompute__(self):
         return _concat, ()
 
     def __dask_postpersist__(self):
         return from_graph, (self._meta, self.divisions, self._name)
 
-
-class Blockwise(Expr):
-    """Super-class for block-wise operations
-
-    This is fairly generic, and includes definitions for `_meta`, `divisions`,
-    `_layer` that are often (but not always) correct.  Mostly this helps us
-    avoid duplication in the future.
-    """
-
-    operation = None
-
-    @property
-    def _meta(self):
-        return self.operation(
-            *[arg._meta if isinstance(arg, Expr) else arg for arg in self.operands]
-        )
+    def __getattr__(self, key):
+        try:
+            # Prioritize `FrameBase` attributes
+            return object.__getattribute__(self, key)
+        except AttributeError as err:
+            try:
+                # Fall back to `expr` API
+                # (Making sure to convert to/from Expr)
+                val = getattr(self.expr, key)
+                if callable(val):
+                    return functools.partial(_wrap_expr_api, wrap_api=val)
+                return val
+            except AttributeError:
+                # Raise original error
+                raise err
```

Comment:
@mrocklin - This is where `FrameBase.__getattr__` falls back to the wrapped version of the `Expr` method. Is this what you were asking about?